caas/kubernetes/provider/modeloperator.go

```diff
@@ -400,6 +502,127 @@ func modelOperatorService(
 	}
 }
 
+func ensureModelOperatorRBAC(
+	broker ModelOperatorBroker,
+	operatorName,
+	namespace string,
+	labels map[string]string,
+) (string, error) {
+	sa := &core.ServiceAccount{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      operatorName,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		AutomountServiceAccountToken: boolPtr(true),
+	}
+
+	if err := broker.EnsureServiceAccount(sa); err != nil {
+		return sa.Name, errors.Annotate(err, "ensuring service account")
+	}
+
+	clusterRole := &rbac.ClusterRole{
+		ObjectMeta: meta.ObjectMeta{
+			Name:   operatorName,
+			Labels: labels,
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"namespaces"},
+				Verbs:     []string{"get"},
+			},
+		},
+	}
+
+	if err := broker.EnsureClusterRole(clusterRole); err != nil {
+		return sa.Name, errors.Annotate(err, "ensuring cluster role")
+	}
+
+	clusterRoleBinding := &rbac.ClusterRoleBinding{
+		ObjectMeta: meta.ObjectMeta{
+			Name:   operatorName,
+			Labels: labels,
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     clusterRole.Name,
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      sa.Name,
+				Namespace: sa.Namespace,
+			},
+		},
+	}
+
+	if err := broker.EnsureClusterRoleBinding(clusterRoleBinding); err != nil {
+		return sa.Name, errors.Annotate(err, "ensuring cluster role binding")
+	}
+
+	role := &rbac.Role{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      operatorName,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{"admissionregistration.k8s.io"},
+				Resources: []string{"mutatingwebhookconfigurations"},
+				Verbs: []string{
+					"create",
+					"delete",
+					"get",
+					"list",
+					"update",
```

Comment:
"patch" as well?
how do we know we have covered everything in the various verb slices?

Reply:
We don't do any patching at the moment in the current code path.